config/wireshark.mpb

```diff
@@ -1,43 +1,60 @@
-project : glib {
-  requires      += wireshark
-
-  // Unfortunately Wireshark does not install headers for plugin
-  // development; instead we rely on the source distribution:
-  includes      += $(WIRESHARK_SRC)
-
-  specific(vc71) {
-    macros      += MSC_VER_REQUIRED=1310
-  }
-  specific(vc8) {
-    macros      += MSC_VER_REQUIRED=1400
-  }
-  specific(vc9) {
-    macros      += MSC_VER_REQUIRED=1500
-  }
-  specific(vc10) {
-    macros      += MSC_VER_REQUIRED=1600
-  }
-  specific(vc11) {
-    macros      += MSC_VER_REQUIRED=1700
-  }
-  specific(vc12) {
-    macros      += MSC_VER_REQUIRED=1800
-  }
-  specific(vc14) {
-    macros      += MSC_VER_REQUIRED=1900
-  }
-  specific(vs2017) {
-    macros      += MSC_VER_REQUIRED=1910
-  }
-  verbatim(nmake, top) {
-    include $(WIRESHARK_SRC)/config.nmake
-  }
-
-  specific(prop:windows) {
-    libpaths    += $(WIRESHARK_SRC)/wiretap $(WIRESHARK_SRC)/epan
-    lit_libs    += wiretap-$(WIRETAP_VERSION) libwireshark
-  } else {
-    libpaths    += $(WIRESHARK_SRC)/wiretap/.libs $(WIRESHARK_SRC)/epan/.libs
-    lit_libs    += wiretap wireshark
-  }
-}
+project : glib {
+  requires      += wireshark_any

```

Comment:
Which features do we expect the user to enable (command line or default.features) and which will be set up in global.features?  In the case where no wireshark is available, what will set wireshark_any to 0?  (See xerces.mpb for an example of supporting two different versions.)

Reply:
Right now I guess nothing does, I'm probably doing this in a way that works on my machine but isn't proper. In configure I add `wireshark_any` to `features` if `wireshark` xor `wireshark2` are defined. [I think this is my current version of configure.](https://github.com/iguessthislldo/OpenDDS/commit/17ab5f8a0c73579d32d4f6d916f2694061a1db3a) I found the features files though and I can add them there later (I have free time at the moment but I'm on the road and all I have is my phone).

Also in xerces.mpb `xerces` is required if 2 or 3 isn't true. Is this what disables the file or does it do something else?

Reply:
Actually I know that version of configure didn't work right because of wireshark2_bin didn't get set correctly because of where it is in the file. I think making it absolute would make everything consistent but I would like it to default to wireshark2_build\run\run\RelWithDebInfo and I'm not sure how that would work.

Reply:
That's fine, we can pick up the discussion later when we're in the same room.

In general we need to make sure that the MPC change works well standalone (without any help from OpenDDS's configure script).

In the xerces example, making the feature "xerces" required means the project will be skipped with a message saying that xerces is required.  Nothing will set xerces to 1 (and it starts as 0 in global.features).  We don't have to make wireshark work the same way but something similar may be useful.

Reply:
Also there is an example in glib.mpb of providing a default value that the user can override using the MPC 'expand' keyword.